test/gtest/uct/test_md.cc

```diff
@@ -989,3 +1007,29 @@ UCS_TEST_P(test_md_memlock_limit, md_open)
 }
 
 UCT_MD_INSTANTIATE_TEST_CASE(test_md_memlock_limit)
+
+class test_md_non_blocking : public test_md
+{
+protected:
+    void init() override {
+        /* ODPv1 IB feature can work only for certain DEVX configuration */
+        modify_config("MLX5_DEVX_OBJECTS", "dct,dcsrq", IGNORE_IF_NOT_EXIST);
+        test_md::init();
+    }
+};
+
+UCS_TEST_SKIP_COND_P(test_md_non_blocking, reg_advise,
+                     !check_caps(UCT_MD_FLAG_REG | UCT_MD_FLAG_ADVISE))
+{
+    test_reg_advise(UCS_KBYTE, UCS_KBYTE, 0, true);
+    test_reg_advise(UCS_KBYTE, UCS_KBYTE / 2, 0, true);
+    test_reg_advise(UCS_KBYTE, UCS_KBYTE / 2, UCS_KBYTE / 4, true);
```

Comment:
why are all the sizes that small? Imo, better to use multiple sizes, so that we could test many pages

Reply:
Due to internal bug that broke ODPv1 in case of huge size and allocation through malloc (not memalign).

Reply:
Which size was it?

Reply:
32K+

All details are here: https://redmine.mellanox.com/issues/3363764 (internal link)

Reply:
can you please add the test with bigger message size, but comment it for now? Also can mention that it needs to be uncomented once this RM is fixed.